### PR TITLE
Fix gadgetron link

### PR DESCRIPTION
**Pull request by paskino**
As described in #709 there are problem resolving libraries at runtime and at build time. This PR tries to address this.

### CODE REVIEW

**Comment by paskino** [comment]
I'm not sure about the `.bashrc`. I think that's a problem but maybe it has nothing to do with this PR.

File: docker/user_sirf-ubuntu.sh [hunk 1]
```diff
@@ -18,6 +18,4 @@ echo $COMPILER_FLAGS
 echo $BUILD_FLAGS $EXTRA_BUILD_FLAGS
 cmake $BUILD_FLAGS $EXTRA_BUILD_FLAGS $COMPILER_FLAGS .
 
-cmake --build . -j ${NUM_PARALLEL_BUILDS} --target Gadgetron
-
-#cmake --build . -j ${NUM_PARALLEL_BUILDS}
+cmake --build . -j ${NUM_PARALLEL_BUILDS} --target
```
Comment (KrisThielemans) [inline docker/user_sirf-ubuntu.sh:21]: empty `--target`?

**Review by KrisThielemans — COMMENTED** [review]
I really don't know if using `USE_SYSTEM_HDF5=OFF` is a good idea. As opposed to having 2 hdf5 builds (`apt` and `conda`), we will now have 3...

Of course, we should be able to safely set `LD_LIBRARY_PATH` to our builds (as opposed to the one from `conda`). However, presumably that means that conda installed code that uses hdf5 will break. Or maybe not, as conda's `python` presumably adds conda lib directories to its path, but then the SIRF build could break...

I think this all points to a serious problem in HDF5 shared libraries. It seems there is no foolproof way around this.

**Comment by KrisThielemans** [comment]
Alternative 1:
------------
create a script (`gadgetron.sh` or even `sirf_gadgetron.sh`) which sets `LD_LIBRARY_PATH` appropriately and runs it. 

pros:
- no extra HDF5 library
- nobody else is affected.

cons:
- you have to remember to call that script
- presumably any SIRF (or STIR) executable that needs HDF5 libraries would need the same treatment, which becomes unmanageable.

Alternative 2:
-------------
build with `USE_SYSTEM_HDF5=on` but make sure gadgetron, STIR and SIRF get linked with the system version.

pros:
- no extra HDF5 library
- no need for `LD_LIBRARY_PATH`

cons:
- not sure what will happen in Python

**Comment by KrisThielemans** [comment]
There's of course

Alternative 3
-------------
use `RPATH` for everything. Could work very well on `docker` as we have complete control.

File: docker/user_sirf-ubuntu.sh (hunk 1 above)
Comment (paskino) [reply docker/user_sirf-ubuntu.sh:21]: wrong!

**Comment by paskino** [comment]
CIL uses `h5py` which in turn uses `HDF5` library. These are installed by conda and get linked with `RPATH`. So, I'd argue that the conda packages will be safe. As a matter of fact, we never had an issue with CIL use of HDF5 with SIRF.

File: docker/user_sirf-ubuntu.sh (hunk 1 above)
Comment (paskino) [reply docker/user_sirf-ubuntu.sh]: fixed now.

**Comment by paskino** [comment]
If `USE_SYSTEM_HDF5=OFF` then the build works fine as we know where the libriaries are.
Option 3 is the best IMO. However, due to time constraints for the PSMRTBP training school, I'd like to merge this as is, possibly not closing the issue?

**Comment by KrisThielemans** [comment]
Sure. Might have to be a squash merge as I guess history is very ugly here

**Comment by paskino** [comment]
I always like to squash n merge

File: SuperBuild/External_Gadgetron.cmake [hunk 2]
```diff
@@ -108,6 +108,7 @@ if(NOT ( DEFINED "USE_SYSTEM_${externalProjName}" AND "${USE_SYSTEM_${externalPr
       -DCMAKE_LIBRARY_PATH:PATH=${SUPERBUILD_INSTALL_DIR}/lib
       -DCMAKE_INCLUDE_PATH:PATH=${SUPERBUILD_INSTALL_DIR}/include
       -DCMAKE_INSTALL_PREFIX:PATH=${Gadgetron_INSTALL_DIR}
+      -DCMAKE_BUILD_RPATH=".;${Gadgetron_INSTALL_DIR}/lib"
```
Comment (KrisThielemans) [inline SuperBuild/External_Gadgetron.cmake:111]: I'm not sure what the consequences of this are, but a `.`  seems terribly dangerous

File: version_config.cmake [hunk 3]
```diff
@@ -211,7 +211,7 @@ else()
 
   ## STIR
   set(DEFAULT_STIR_URL https://github.com/UCL/STIR )
-  set(DEFAULT_STIR_TAG rel_5.0.1)
+  set(DEFAULT_STIR_TAG release_5.0)
```
Comment (KrisThielemans) [inline version_config.cmake:214]: shouldn't be in this PR Note that 5.0.2 is out, so better to use that in a separate PR. see https://github.com/SyneRBI/SIRF-SuperBuild/pull/719

**Review by KrisThielemans — CHANGES_REQUESTED** [review]

**Comment by KrisThielemans** [comment]
I believe this is obsolete and should be abandoned

**Comment by KrisThielemans** [comment]
Obsolete?

**Comment by paskino** [comment]
Yes obsolete, closing

**Pull request closed without merging**